cmd/thanos/compact.go

```diff
@@ -388,12 +388,24 @@ func runCompact(
 	}
 
 	if retentionByResolution[compact.ResolutionLevelRaw].Seconds() != 0 {
+		// If downsampling is enabled, error if raw retention is not sufficient for downsampling to occur (upper bound 10 days for 1h resolution)
+		if !conf.disableDownsampling && retentionByResolution[compact.ResolutionLevelRaw].Seconds() < downsample.ResLevel1DownsampleRange {
+			return errors.New("raw resolution must be higher than the minimum block size after which 1h resolution downsampling will occur (40 hours)")
```

Comment:
I don't get why raw block retention is related to 1h downsampled block. I think 1h downsampled block is created from 5m downsampled block. Can you please explain it?

Reply:
Copy pasta typo, fixing.